solutions/swb-app/src/dataSetRoutes.ts

```diff
@@ -105,6 +109,34 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
     })
   );
 
+  // Add file to dataset
+  router.get(
+    '/datasets/:dataSetId/upload-requests',
+    wrapAsync(async (req: Request, res: Response) => {
+      if (req.params.dataSetId.match(uuidWithLowercasePrefixRegExp(resourceTypeToKey.dataset)) === null) {
+        throw Boom.badRequest('dataSetId request parameter is invalid');
+      }
+
+      const validatedRequest = validateAndParse<DataSetFileUploadRequest>(DataSetFileUploadRequestParser, {
+        dataSetId: req.params.dataSetId,
+        filenames: req.query.filenames
+      });
+      const authenticatedUser = validateAndParse<AuthenticatedUser>(AuthenticatedUserParser, res.locals.user);
+
+      if (typeof validatedRequest.filenames === 'string') {
```

Comment:
is there a reason we don't want this to always be an array even if its a single entry?